editor/plugins/path_2d_editor_plugin.cpp

```diff
@@ -475,9 +485,28 @@ void Path2DEditor::_mode_selected(int p_mode) {
 		curve_edit->set_pressed(false);
 		curve_edit_curve->set_pressed(false);
 		curve_del->set_pressed(true);
-	} else if (p_mode == ACTION_CLOSE) {
-		//?
+	} else if (p_mode == MODE_CLEAR_POINTS) {
+		if (!node->get_curve().is_valid()) {
+			return;
+		}
+		if (node->get_curve()->get_point_count() == 0) {
```

Comment:
Ideally the button should be disabled when there are no points, not sure if it's easy to implement.

Reply:
Yeah, the problem is that if you want to implement this, you would have to enable the button when a new curve is created or removed as well. I even imagine that Curve2D could emit a signal to path_2d_editor_plugin.cpp every time a new curve is added/removed. And this would apply not only to the clear point button, but to the other buttons as well (edit, create, remove, etc). 

This way, this would be related to the functionality of each button, in the case of remove (point), only enable it if the curve is valid and if there is at least a single point to be deleted. Something similar to the clear point.

In my opinion, this is not that difficult to be made, but it should be implemented in another pr.

Reply:
I was thinking if the right thing would be to save which mode each node (Path2D) selected.